service/src/main.rs

```diff
@@ -162,157 +184,189 @@ impl Inclusion for Arc<InclusionService> {
+                                )?;
+                                Ok::<
+                                    (),
+                                    sled::transaction::ConflictableTransactionError<
+                                        InclusionServiceError,
+                                    >,
+                                >(())
+                            })
+                            .map_err(|e| InclusionServiceError::GeneralError(e.to_string()))?;
+                    }
+                    JobStatus::ZkProofFinished(_) => (),
+                    JobStatus::Failed(_) => {
+                        // TODO: "Need to impl some way to retry some failures, and report perminent failures here"
+                        ()
+                    }
+                }
+            }
         }
+        Ok(None)
+    }
+
+    fn insert_job_status_to_queue(
+        &self,
+        job_key: Vec<u8>,
+        job_status: JobStatus,
+        job: Job,
+    ) -> Result<(), InclusionServiceError> {
+        self.queue_db
+            .insert(
+                job_key,
+                bincode::serialize(&job_status)
+                    .map_err(|e| InclusionServiceError::GeneralError(e.to_string()))?,
+            )
+            .map_err(|e| InclusionServiceError::GeneralError(e.to_string()))?;
+        Ok(self
+            .job_sender
+            .send(job)
+            .map_err(|e| InclusionServiceError::GeneralError(e.to_string()))?)
     }
 }
 
-async fn prove(
-    job: Job,
+/// Connect to the Cestia [`Client`] and attempt to get a NMP for a [`Job`].
+/// A successful Result indicates that the queue DB contains valid ZKP input
+async fn get_zk_proof_input_from_da(
+    job: &Job,
     client: Arc<Client>,
-    queue_tree: SledTree,
-    proof_tree: SledTree,
-) -> Result<(), InclusionServiceError> {
+) -> Result<KeccakInclusionToDataRootProofInput, InclusionServiceError> {
+    debug!("Preparing request to Celestia...");
+    let height = job.height;
+
+    let commitment =
+        Commitment::new(job.commitment.clone().try_into().map_err(|_| {
+            InclusionServiceError::InvalidParameter("Invalid commitment".to_string())
+        })?);
+
+    let namespace = Namespace::new_v0(&job.namespace).map_err(|e| {
+        InclusionServiceError::InvalidParameter(format!("Invalid namespace: {}", e))
+    })?;
+
+    debug!("Getting blob from Celestia...");
+    let try_blob = client.blob_get(height, namespace, commitment).await;
+    let blob = try_blob
+        .inspect_err(|e: &ClientError| {
+            match e {
+                ClientError::Call(error_object) => {
+                    // TODO: make this handle errors much better!  See ErrorCode::ServerError(1){
+                    if error_object.message() == "header: not found" {
+                        todo!();
+                    };
+                }
+                // TODO: handle other Celestia JSON RPC errors
+                _ => (),
+            }
+        })
```

Comment:
WIP hook to handle specific JSON RPC errors and decide if the are recoverable/retryable or not

Reply:
got a better but still very messy thing to replace this in place in https://github.com/celestiaorg/eq-service/pull/20/commits/117c732307979c570736fd922d72d89d00bc7377